Core/Core/Use Cases/GetSubmission.swift

```diff
@@ -43,25 +39,22 @@ public class GetSubmission: APIUseCase {
     }
 
     public var scope: Scope {
-        return Scope(predicate: mainPredicate, order: [mainSort])
+        return Scope(
+            predicate: NSPredicate(
+                format: "%K == %@ AND %K == %@",
+                #keyPath(Submission.assignmentID),
+                assignmentID,
+                #keyPath(Submission.userID),
+                userID
+            ),
+            order: [NSSortDescriptor(key: #keyPath(Submission.attempt), ascending: false)]
+        )
     }
 
     public func write(response: APISubmission?, urlResponse: URLResponse?, to client: PersistenceClient) throws {
-        guard let item = response else {
+        guard let item = response, let context = client as? NSManagedObjectContext else {
             return
         }
-
-        let model: Submission = client.fetch(predicate: mainPredicate, sortDescriptors: [ mainSort ]).first ?? client.insert()
-        try model.update(fromApiModel: item, in: client)
-        for entry in item.submission_history ?? [] {
-            let predicate = NSPredicate(
-                format: "%K == %@ AND %K == %@ AND %K == %d",
-                #keyPath(Submission.assignmentID), assignmentID,
-                #keyPath(Submission.userID), userID,
-                #keyPath(Submission.attempt), entry.attempt ?? 0
-            )
-            let historical: Submission = client.fetch(predicate: predicate, sortDescriptors: nil).first ?? client.insert()
-            try historical.update(fromApiModel: entry, in: client)
-        }
+        Submission.save(item, in: context)
```

Comment:
can `Submission` be a `WriteableModel`, and use the default `write`?

Reply:
I think it could after pulling Persistence out since right now the default `write` uses `PersistenceClient` and not `NSManagedObjectContext`. We could do the down cast inside the model but I'd rather not.

Reply:
Nevermind, I was just looking at some code and I think I understand now. Should be doable.